File: 프로그래머스/2/42578. 의상/의상.java
import java.util.*;

class Solution {
    public int solution(String[][] clothes) {
        HashMap<String, Integer> map = new HashMap<>();
        
        int answer = 1;
        int length = clothes.length;
        
        for(int i=0; i<length; i++) {
            map.put(clothes[i][1], map.getOrDefault(clothes[i][1], 0) + 1);
        }
        
        for(String key : map.keySet()) {
            answer *= (map.get(key) + 1);
        }
        return answer - 1;
        
    }
}

File: 프로그래머스/2/42578. 의상/의상.js
function solution(clothes) {
    let map = new Map();
    for(let cloth of clothes) {
        map.set(cloth[1], (map.get(cloth[1]) || 0) + 1)
    };
        
    let answer = 1;
    for(let [key, value] of map) {
        answer *= value + 1
    }
        
    return answer - 1
}